atlasdb-client/src/main/java/com/palantir/atlasdb/cache/TimestampStore.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cache;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.palantir.atlasdb.persistent.api.LogicalPersistentStore;
+import com.palantir.atlasdb.persistent.api.PersistentStore;
+import com.palantir.atlasdb.persistent.api.PersistentStore.StoreNamespace;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.common.streams.KeyedStream;
+
+/**
+ * Stores timestamps using delta encoding for commit timestamp.
+ */
+public class TimestampStore implements LogicalPersistentStore<Long, Long> {
+    private final PersistentStore physicalPersistentStore;
```

Comment:
If you're making the distinction between logical and physical, then the physical one should actually be `PhysicalPersistentStore`.

Reply:
Renamed.